axelrod/tests/unit/test_filters.py

```diff
@@ -0,0 +1,155 @@
+        self.assertTrue(passes_in_list_filter(
+            self.TestStrategy, 'makes_use_of', 'length'))
+        self.assertFalse(passes_in_list_filter(
+            self.TestStrategy, 'makes_use_of', 'test'))
+
+    @given(
+        truthy=sampled_from([True, 'True', 'true', '1', 'Yes', 'yes']),
+        falsy=sampled_from([False, 'False', 'false', '0', 'No', 'no']),
+        smaller=integers(min_value=0, max_value=9),
+        larger=integers(min_value=11, max_value=100),
+    )
+    @example(truthy=True, falsy=False, smaller='2', larger=float('inf'))
+    def test_passes_filterset(self, truthy, falsy, smaller, larger):
+
+        full_passing_filterset = {
+            'stochastic': truthy,
+            'inspects_source': falsy,
+            'min_memory_depth': smaller,
+            'max_memory_depth': larger,
+            'makes_use_of': 'length'
+        }
+
+        sparse_passing_filterset = {
+            'stochastic': truthy,
+            'inspects_source': falsy,
+            'makes_use_of': 'length'
+        }
+
+        full_failing_filterset = {
+            'stochastic': falsy,
+            'inspects_source': falsy,
+            'min_memory_depth': smaller,
+            'max_memory_depth': larger,
+            'makes_use_of': 'length'
+        }
+
+        sparse_failing_filterset = {
+            'stochastic': falsy,
+            'inspects_source': falsy,
+            'min_memory_depth': smaller,
+        }
+
+        self.assertTrue(passes_filterset(
+            self.TestStrategy, full_passing_filterset))
+        self.assertTrue(passes_filterset(
+            self.TestStrategy, sparse_passing_filterset))
+        self.assertFalse(passes_filterset(
+            self.TestStrategy, full_failing_filterset))
+        self.assertFalse(passes_filterset(
+            self.TestStrategy, sparse_failing_filterset))
+
+    def test_filtered_strategies(self):
+
+        class StochasticTestStrategy(object):
+            classifier = {
+                'stochastic': True,
+                'memory_depth': float('inf'),
+                'makes_use_of': []
+            }
+
+        class MemoryDepth2TestStrategy(object):
+            classifier = {
+                'stochastic': False,
+                'memory_depth': 2,
+                'makes_use_of': []
+            }
+
+        class UsesLengthTestStrategy(object):
+            classifier = {
+                'stochastic': True,
+                'memory_depth': float('inf'),
+                'makes_use_of': ['length']
+            }
+
+        strategies = [
+            StochasticTestStrategy,
+            MemoryDepth2TestStrategy,
+            UsesLengthTestStrategy
+        ]
+
+        stochastic_filterset = {
+            'stochastic': True
+        }
+
+        deterministic_filterset = {
+            'stochastic': False
+        }
+
+        uses_length_filterset = {
+            'stochastic': True,
+            'makes_use_of': 'length'
+        }
+
+        self.assertEqual(
```

Comment:
Could you also include some tests that show using the filter gives the same results as using some list comprehensions? (This might be more helpful to me right now just so that I see how it all works).